### PR TITLE
Added Database initialization and some config management

### PR DESCRIPTION
**Pull request by Wojtechnology**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+)
+
+// Environment variable keys
+const (
+	environmentKey = "ITERABLE_ENVIRONMENT"
+	configDirKey   = "ITERABLE_CONFIG_DIR"
+)
+
+// Other constants
+const (
```
Comment (rossem) [inline config/config.go]: dont need ( )

File: config/config.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		return err
+	}
+	if err := configParserFunc(data, conf); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get config file path from environment variables
+func getConfigPath() string {
```
Comment (rossem) [inline config/config.go:57]: Can this error? What if env vars arent set

File: main.go [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
+import (
+	"fmt"
+
+	_ "github.com/iterableio/api/config"
+	"github.com/iterableio/api/db"
+)
 
 func main() {
-	fmt.Printf("Hello World.\n")
+	fmt.Printf("Starting db\n")
```
Comment (narensrini) [inline main.go]: Any reason we use `Printf` rather than `Println`

Comment (rossem) [inline main.go]: Also we should prob use log, it just gives more info

File: config/development.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+postgres:
```
Comment (narensrini) [inline config/development.yaml:1]: Pretty nitpicky, but I thought the file extension is usually `.yml`?

File: config/test.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+postgres:
+  user: postgres
+  password: postgres
+  dbname: iterable_test
```
Comment (rossem) [inline config/test.yaml:4]: So we also need our migrations to run on iterable_test? @nare469 do you know how to do this? can we have db/migrate_test

File: config/config.go (hunk 2 above)
Comment (Wojtechnology) [inline config/config.go:57]: Nope, defaults to empty string in which case something else will error.

File: main.go (hunk 3 above)
Comment (Wojtechnology) [inline main.go]: This is temp

File: config/development.yaml (hunk 4 above)
Comment (Wojtechnology) [inline config/development.yaml:1]: It's actually `.yaml`

File: config/test.yaml (hunk 5 above)
Comment (Wojtechnology) [inline config/test.yaml:4]: I think you can add a `TestMain` to the db_test file. Look here for more info https://golang.org/pkg/testing/.

File: main.go (hunk 3 above)
Comment (rossem) [inline main.go]: I don't think its a bad thing to have permanently

Comment (Wojtechnology) [inline main.go]: We'll probably need to come up with better logs when you merge your http stuff. For now it's fine.

File: config/test.yaml (hunk 5 above)
Comment (narensrini) [inline config/test.yaml:4]: Ugh. I'm definitely gonna have to make a bunch of changes to db/migrate after this PR gets in, but yeah. db/migrate_test sounds good.
Also, what's the point of having a test db?

File: main.go [hunk 6]
```diff
@@ -1,7 +1,14 @@
 package main
 
-import "fmt"
+import (
+	"fmt"
+
+	_ "github.com/iterableio/api/config"
```
Comment (rossem) [inline main.go:6]: minor, but can we put all the `_` at the bottom

File: db/db.go [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+
+	"github.com/iterableio/api/config"
+)
+
+var DB *sqlx.DB
+
+func ConnectSQL() error {
```
Comment (rossem) [inline db/db.go:14]: so whats the advantage of not having this shit be inside `init`? I feel like this can be cleaner, as the connection to the db will be self contained and the other packages dont have to worry about whether its connected or not, they will know for sure that it is

File: db/db.go [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+	var err error
+	DB, err = sqlx.Connect("postgres", buildDataSourceName())
+	if err != nil {
```
Comment (rossem) [inline db/db.go]: you can just `return err`

File: db/db_test.go [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildDataSourceName(t *testing.T) {
+	assert := assert.New(t)
+
+	dataSourceName := buildDataSourceName()
+
+	assert.Equal(dataSourceName,
```
Comment (rossem) [inline db/db_test.go:17]: hm, can we actually use the config here to test this? what if we change something in config, then we will have to change it here as well

File: db/db_test.go [hunk 10]
```diff
@@ -0,0 +1,25 @@
+package db
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildDataSourceName(t *testing.T) {
+	assert := assert.New(t)
```
Comment (rossem) [inline db/db_test.go:13]: why is this needed

File: config/test.yaml (hunk 5 above)
Comment (rossem) [inline config/test.yaml:4]: Yeah, not sure, I think we can just have make test run a db reset and then test

**Comment by rossem** [comment]
Nice work, just some minor changes/questions from me

File: config/test.yaml (hunk 5 above)
Comment (Wojtechnology) [inline config/test.yaml:4]: I like that ^

File: main.go (hunk 6 above)
Comment (Wojtechnology) [inline main.go:6]: Go will automatically alphabetize. I kind of like the way it is now. Better to group by source of lib rather than whether it's used or not.

File: db/db.go (hunk 7 above)
Comment (Wojtechnology) [inline db/db.go:14]: It's more to be explicit and have control over the order that things are initiated in.

File: db/db_test.go (hunk 10 above)
Comment (Wojtechnology) [inline db/db_test.go:13]: It's nice

Comment (rossem) [inline db/db_test.go:13]: yes but what does it do?

Comment (Wojtechnology) [inline db/db_test.go:13]: It's an assertion library with pretty printing

Comment (rossem) [inline db/db_test.go:13]: Yeah ik lmao, im wondering why that line is needed, i thought you could just do assert.Equal right away

Comment (Wojtechnology) [inline db/db_test.go:13]: That's how they did it in examples, I think you need to pass in the test object so that it can halt operation if an assertion fails.

**Comment by rossem** [comment]
LGTM, can you put vim stuff in gitignore and then merge
